### PR TITLE
Added validation of analytic type

### LINKED ISSUE #2257: Analytics no validation when "Type" is null
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.2.0-RC4
| Browser Version |Chromium

#### Actual Behavior

If you enter a title and hit "OK" the dialog closes and nothing is "added":

getzne5

#### Expected Behavior

I would expect an error message.

#### Steps to Reproduce

Click the add button, enter a title, hit "OK".
screenshots or links to GitHub repositories that reproduce the problem.

#### Possible Solutions

If you have already ideas how to solve the issue, add them here.
(remove this section if not needed)

### LINKED ISSUE #2262: CustomUrl: No AJAX spinner whilst saving.
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | Specific version or SHA of a commit
| Browser Version | Browser name and version

#### Actual Behavior

When you click save there is no indication in the UI that the save is being performed. The
case is special as this could be the only overlay in Sulu which performs a large AJAX request.

sulu2

#### Expected Behavior

I would expect some UI state change to happen when I click "save", and maybe an AJAX spinner to appear somewhere.

#### Possible Solutions

- Animate the save button?

/cc @berndhep

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #2257 , fixes #2262 , #2260
| Related issues/PRs | none
| License | MIT
| Documentation PR | none 

#### What's in this PR?

This PR adds validation of analytics type.

### CODE REVIEW

File: src/Sulu/Component/CustomUrl/Generator/Generator.php [hunk 1]
```diff
@@ -48,6 +48,10 @@ public function generate($baseDomain, $domainParts, Localization $locale = null)
         }
 
         foreach ($domainParts['suffix'] as $suffix) {
+            if (empty($suffix)) {
+                continue;
+            }
```
Comment (danrot) [inline src/Sulu/Component/CustomUrl/Generator/Generator.php]: Why was this change required?

Comment (wachterjohannes) [inline src/Sulu/Component/CustomUrl/Generator/Generator.php]: because you could "push" empty parts into the api which is quite bad and should not work.

Comment (danrot) [inline src/Sulu/Component/CustomUrl/Generator/Generator.php]: Ok, any real world damage caused by that, or is it just some detail?
